contrib/pod-master/podmaster.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// podmaster is a simple utility, it attempts to acquire and maintain a lease-lock from etcd using compare-and-swap.
+// if it is the master, it copies a source file into a destination file.  If it is not the master, it makes sure it is removed.
+//
+// typical usage is to copy a Pod manifest from a staging directory into the kubelet's directory, for example:
+//   podmaster --etcd-servers=http://127.0.0.1:4001 --key=scheduler --source-file=/kubernetes/kube-scheduler.manifest --dest-file=/manifests/kube-scheduler.manifest
+
```

Comment:
to get the above to show up in godoc, there can't be blank lines between the comment and the package declaration.